onnxruntime/test/fuzzing/src/OnnxPrediction.cpp

```diff
@@ -2,40 +2,57 @@
 // Licensed under the MIT License.
 
 #include "OnnxPrediction.h"
+#include <vector>
+#include <memory>
 
 // Uses the onnxruntime to load the model
 // into a session.
 //
 OnnxPrediction::OnnxPrediction(std::wstring& onnx_model_file)
-:
-rawModel{nullptr},
-ptrSession{nullptr},
-session{ env, onnx_model_file.c_str(), emptySessionOption},
-inputNames{session.GetInputCount()},
-outputNames{session.GetOutputCount()}
-{
+    : rawModel{nullptr},
+      ptrSession{nullptr},
+      session{env, onnx_model_file.c_str(), emptySessionOption},
+      inputNames{session.GetInputCount()},
+      outputNames{session.GetOutputCount()} {
   init();
 }
 
 // Uses the onnx to seri
 //
 OnnxPrediction::OnnxPrediction(onnx::ModelProto& onnx_model)
-:
-session{nullptr}
-{
-  rawModel = std::shared_ptr<void>{alloc.Alloc(onnx_model.ByteSizeLong()),   
-  [this](void * ptr)
-      { 
-      this->GetAllocator().Free(ptr); 
-      }
-  };
+    : session{nullptr} {
+  rawModel = std::shared_ptr<void>{alloc.Alloc(onnx_model.ByteSizeLong()),
+                                   [this](void* ptr) {
+                                     this->GetAllocator().Free(ptr);
+                                   }};
 
   onnx_model.SerializeToArray(rawModel.get(), static_cast<int>(onnx_model.ByteSizeLong()));
 
-  ptrSession = std::make_unique<Ort::Session>(env, 
-      rawModel.get(), 
-      onnx_model.ByteSizeLong(), 
-      emptySessionOption),
+  ptrSession = std::make_unique<Ort::Session>(env,
+                                              rawModel.get(),
+                                              onnx_model.ByteSizeLong(),
+                                              emptySessionOption),
+
+  inputNames.resize(ptrSession->GetInputCount());
+  outputNames.resize(ptrSession->GetOutputCount());
+
+  init();
+}
+
+OnnxPrediction::OnnxPrediction(const std::vector<char>& modelData)
+    : numBytes(modelData.size()),
```

Comment:
>numBytes [](start = 6, length = 8)

Seems this is only used in the constructor, do you need this to be a member variable?

Reply:
That is correct. I had some other changes which I didn't commit, that required the numBytes as a member. I will remove from this commit.